Remove redundant ESP32 board

marekmatej opened the pull request:
This PR merges two similar boards `esp32_devkitc_wroom` and `esp32_devkitc_wrover` into a single `esp32_devkitc` board. The merged boards were used to split PSRAM and non-PSRAM variants. But since the configuration of the PSRAM is quite board independent, it starts to be clear these boards are almost identical. The unification also lowers the number of tested boards so it can also lower down the CI execution time.

marekmatej committed:
boards: esp32: remove redundant board

Remove the esp32_devkitc_wroom board as near identical with wrover
variant. The differences of the two boards are covered by the user config.

Signed-off-by: Marek Matej <marek.matej@espressif.com>

marekmatej committed:
boards: esp32: make only one esp32_devkitc

Rename the esp32_devikic_wrover to esp32_devkitc.

Signed-off-by: Marek Matej <marek.matej@espressif.com>

marekmatej committed:
tests, samples: update esp32_devkitc board records

Update the tests using the esp32_devkitc_xxx entries.

Signed-off-by: Marek Matej <marek.matej@espressif.com>

nordicjm reviewed (dismissed):
Should add deprecated entries for the removed boards

marekmatej committed:
doc,boards: mark deprecated boards

Add esp32_devkitc_wroom and esp32_devkitc_wrover on deprecated list.
Make the note on the merge in the migration guide for Zephyr v4.2.

Signed-off-by: Marek Matej <marek.matej@espressif.com>

marekmatej commented:
> Should add deprecated entries for the removed boards

PTAL @nordicjm

kartben commented:
@marekmatej note that you also have CI failures

marekmatej commented:
> @marekmatej note that you also have CI failures

Would you please re-run the CI, it looks like twister scripts are failing for some reason.

marekmatej commented on boards/deprecated.cmake:
oops :) sure!

kartben commented on boards/espressif/esp32_devkitc/doc/index.rst:
```diff
@@ -8,7 +8,7 @@ with integrated Wi-Fi & dual-mode Bluetooth. The ESP32 series employs a
 variations. ESP32 is created and developed by Espressif Systems, a
 Shanghai-based Chinese company, and is manufactured by TSMC using their 40nm
-process. For more information, check `ESP32-DevKitC-WROVER`_.
+process. For more information, check `ESP32-DevKitC`_.
```
Need to update L252 accordingly

marekmatej replied:
thx, the link target needs to be updated too.

kartben commented:
@marekmatej 

```
ERROR   - integration_platforms in drivers.clock.clock_control_api_default - unrecognized platform - esp32_devkitc_wroom/esp32/procpu
```

kartben commented:
```
ERROR   - platform_allow in sample.board.esp32.flash_memory_mapped - unrecognized platform - esp32_devkitc_wroom/esp32/procpu
```

@marekmatej i would recommend a more systematic search and replace :) Thanks!

marekmatej commented:
> ```
> ERROR   - platform_allow in sample.board.esp32.flash_memory_mapped - unrecognized platform - esp32_devkitc_wroom/esp32/procpu
> ```
> 
> @marekmatej i would recommend a more systematic search and replace :) Thanks!

yes, I used an insufficient grep pattern :)

kartben commented on snippets/mcuboot.bak/README.rst:
@marekmatej this looks very wrong! can you please follow up with a PR removing this ASAP?
